File: ADA/Laboratorio05/Ejercicio2.java
// Estudiante: Gustavo Eduardo ORDOÑO POMA
// CUI: 20213036
// Laboratorio05 - Técnicas de Diseño de Algoritmos - Algoritmos Voraces
// Fecha: 01/11/2022

public class Ejercicio2 {
    public static void main(final String[] args) {
        GraphMatrix g = new GraphMatrix(false, 9, false, true);
        g.addEdge(1, 2, 4, true);
        g.addEdge(1, 8, 9, true);
        g.addEdge(2, 3, 9, true);
        g.addEdge(2, 8, 11, true);
        g.addEdge(3, 4, 7, true);
        g.addEdge(3, 6, 4, true);
        g.addEdge(3, 9, 2, true);
        g.addEdge(4, 5, 10, true);
        g.addEdge(4, 6, 15, true);
        g.addEdge(5, 6, 11, true);
        g.addEdge(6, 7, 2, true);
        g.addEdge(7, 8, 1, true);
        g.addEdge(7, 9, 6, true);
        g.addEdge(8, 9, 7, true);

        System.out.println("Generated Matrix:");
        System.out.println("Matrix Representation:");
        g.showMatrixRepresentation();
        System.out.println("Linking Representation:");
        g.showLinkingRepresentation();
        System.out.println();
    
        GraphMatrix MST = g.kruskal();
        System.out.println("Minimum Spanning Tree (MST):");
        System.out.println("Matrix Representation:");
        MST.showMatrixRepresentation();
        System.out.println("Linking Representation:");
        MST.showLinkingRepresentation();
        System.out.println();
        System.out.print("Minimum Cost Possible: "+MST.getAccumulatedWeight());
    }
}

File: ADA/Laboratorio01/Ejercicio2.cpp
// Estudiante: Gustavo Eduardo ORDOÑO POMA
// CUI: 20213036
// Laboratorio01 - Introducción
// Fecha: 27/09/2022

#include <iostream>

using namespace std;

void showPositions(int amountNums, int nums[]) {
    for(int i = 0; i < amountNums; i++) {
        cout << i << " -> " << nums[i] << endl;
    }
}

int main() {
    int amountNums;
    cout << "Cantidad de Elementos: ";
    cin >> amountNums;

    int nums[amountNums];
    for(int i = 0; i < amountNums; i++) {
        cout << "Elemento #" << (i+1) << ": ";
        cin >> nums[i];
    }
 
    showPositions(amountNums, nums);

    return 0;
}